x-pack/platform/plugins/private/observability_ai_assistant_management/public/routes/components/settings_tab/product_doc_entry.tsx

```diff
@@ -42,10 +41,19 @@ export function ProductDocEntry() {
 
   useEffect(() => {
     if (isStatusLoading) return;
-    if (status) {
-      setInstalled(status.overall === 'installed' && status.inferenceId === selectedInferenceId);
+    if (status && status.inferenceId === selectedInferenceId) {
+      if (status.overall === 'installing') {
+        setInstalling(true);
+        setInstalled(false);
+      } else if (status.overall === 'installed') {
+        setInstalling(false);
+        setInstalled(true);
+      } else if (status.overall === 'uninstalled') {
+        setInstalling(false);
+        setInstalled(false);
```

Comment:
This looks odd?

Reply:
This whole file is weird (I know that's not you). We have `isInstalling`, `isInstalled`, `isStatusLoading`. I would expect something like this in terms of state management:

```
installStatusFetch (inferenceEndpointId) {
  loading: boolean,
  installStatus: 'notInstalled' | 'installing' | 'installed' | 'failed'
}
```

and everything to be managed via one hook. e.g. `useGetProductStatus` needs to know when the product docs have been uninstalled or installed. So you'd have something like `useProductDocs(knowledgeBase.inferenceEndpointId)`.